apps/versions/tests.py

```diff
@@ -515,6 +516,27 @@ def test_version_link(self):
         eq_(link, reverse('addons.versions', args=[addon.slug, version]))
         eq_(doc('.version').attr('id'), 'version-%s' % version)
 
+    def test_status_public_only(self):
+        doc = self.get_content(status='public')
+        version = self.addon.current_version.version
+        assert doc('.version').attr('id') == 'version-%s' % version
+
+    def test_bogus_status_listing_all(self):
+        doc = self.get_content(status='foobardummy')
+        version = self.addon.current_version.version
+        assert doc('.version').attr('id') == 'version-%s' % version
+
+    def test_status_beta_without_beta_builds(self):
+        doc = self.get_content(status='beta')
+        assert len(doc('.version')) == 0
+
+    def test_status_beta_with_beta_builds(self):
+        qs = File.objects.filter(version=self.addon.current_version)
+        qs.update(status=amo.STATUS_BETA)
+        doc = self.get_content(status='beta')
+        version = self.addon.current_version.version
+        assert doc('.version').attr('id') == 'version-%s' % version
```

Comment:
I guess figuring out how to add one more version in this test is unavoidable after all :-)